package/Dockerfile

```diff
@@ -7,19 +7,20 @@ RUN zypper -n update && \
     zypper -n clean -a && \
     rm -rf /tmp/* /var/tmp/* /usr/share/doc/packages/*
 
-RUN useradd -u 1000 machine
+RUN useradd -m -u 1000 machine
 
-RUN mkdir -p .docker/machine/machines /etc/rancher/ssl /home/machine && \
-    chown -R machine /etc/rancher/ssl && \
-    chown -R machine /home/machine
+RUN mkdir -p .docker/machine/machines /etc/rancher/ssl /home/machine/bin && \
+    chown -R machine /etc/rancher/ssl
 
-COPY download_driver.sh /usr/local/bin/
-RUN chmod +x /usr/local/bin/download_driver.sh
+COPY download_driver.sh rancher-machine entrypoint.sh /home/machine/bin/
 
-COPY rancher-machine entrypoint.sh /usr/local/bin/
-RUN chmod 0777 /usr/local/bin
+RUN chmod -R 700 /home/machine/bin && \
+    chown -R machine /home/machine
```

Comment:
The homedir itselt was created on L10. I can only see the bin dir as potentially having a different owner so you can scope the chown to it:
```suggestion
    chown -R machine /home/machine/bin
```

Reply:
@pjbgf it was more of a precautionary/general scope to make sure that anything standing within `/home/machine` is owned by `machine`by default (in the future, if another folder/script is added under `/home/machine`, it will also be covered). 

@macedogm I think we should keep both `chmod -R 700 /home/machine/bin && chown -R machine /home/machine`, as the [COPY](https://github.com/rancher/machine/blob/b03f0728096c758531f13f279ea54810212857fb/package/Dockerfile#L15) of the scripts (not shown in the below screenshot) is performed as `root` and assigns `root:root` and `751`on the files.

Here is a small test I performed earlier to analyze the permissions set through the Dockerfile execution, before and after the `COPY` of the scripts:

![image](https://github.com/rancher/machine/assets/12878731/8f2294c1-b43b-489a-ba31-fb373c2f7a31)